plugins/src/lib.rs

```diff
@@ -172,27 +172,29 @@ impl Parse for DeriveSerde {
                 );
                 continue;
             }
-            match meta.lit {
-                Lit::Bool(LitBool { value: true, .. }) if cfg!(feature = "serde1") => {
-                    result = result.and(Ok(Some(true)))
-                }
-                Lit::Bool(LitBool { value: true, .. }) => {
-                    extend_errors!(
+            if let Expr::Lit(expr_lit) = &meta.value {
+                match expr_lit.lit {
+                    Lit::Bool(LitBool { value: true, .. }) if cfg!(feature = "serde1") => {
+                        result = result.and(Ok(Some(true)))
+                    }
+                    Lit::Bool(LitBool { value: true, .. }) => {
+                        extend_errors!(
+                            result,
+                            syn::Error::new(
+                                meta.span(),
+                                "To enable serde, first enable the `serde1` feature of tarpc"
+                            )
+                        );
+                    }
+                    Lit::Bool(LitBool { value: false, .. }) => result = result.and(Ok(Some(false))),
```

Comment:
Here's a refactoring that catches the non-literal error:

```suggestion
            let Expr::Lit(expr_lit) = &meta.value else {
                extend_errors!(
                    result,
                    syn::Error::new(
                        meta.value.span(),
                        "`derive_serde` expects a literal of type `bool`"
                    )
                );
                continue;
            };
            match expr_lit.lit {
                Lit::Bool(LitBool { value: true, .. }) if cfg!(feature = "serde1") => {
                    result = result.and(Ok(Some(true)))
                }
                Lit::Bool(LitBool { value: true, .. }) => {
                    extend_errors!(
                        result,
                        syn::Error::new(
                            meta.span(),
                            "To enable serde, first enable the `serde1` feature of tarpc"
                        )
                    );
                }
                Lit::Bool(LitBool { value: false, .. }) => result = result.and(Ok(Some(false))),
```